File: java/gradle-build-logic/java-plugins/src/main/java/software/onepiece/gradle/pluginshowto/MyJavaApplicationPlugin.java
package software.onepiece.gradle.pluginshowto;

import org.gradle.api.NonNullApi;
import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.gradle.api.plugins.ApplicationPlugin;

@NonNullApi
abstract public class MyJavaApplicationPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
    }
}

File: clojure/gradle-build-logic/java-plugins/src/main/clojure/software/onepiece/gradle/pluginshowto/MyJavaApplicationPlugin.clj
(ns software.onepiece.gradle.pluginshowto.MyJavaApplicationPlugin
    (:import
      (org.gradle.api Project Plugin)
      (org.gradle.api.plugins ApplicationPlugin)
    )
    (:gen-class
      :name       software.onepiece.gradle.pluginshowto.MyJavaApplicationPlugin
      :implements [org.gradle.api.Plugin])
    )

(defn -apply [this ^org.gradle.api.Project project]
  (do
    (.apply (.getPlugins project) "software.onepiece.gradle.pluginshowto.java-base")
    (.apply (.getPlugins project) ApplicationPlugin)
  )
)
